File: 3_mvcCore/code/3aspnetmvcefapp/Models/Employee.cs
using System.ComponentModel.DataAnnotations;
namespace Aspnetmvcapp
{
    public class Employee
    {
        public int ID{get;set;}
        [Required]
        [StringLength(25)]
        [MinLength(3,ErrorMessage="First Name must be more than 3 char")]
        public string FirstName{get;set;}

        [Required]
        [StringLength(25)]
        [MinLength(3,ErrorMessage="Last Name must be more than 3 char")] 
        public string LastName{get;set;}

        [Range(10000,500000)]
        public int Salary{get;set;}
    }
}

File: 3_mvcCore/code/vb/FirstWebAPI/FirstWebAPI/FirstWebAPI/Models/Employee.vb
Imports System.ComponentModel.DataAnnotations

Public Class Employee
    Private id As String
    Private city As String
    Private salary As Integer
    Private name As String
    <Key>
    Public Property EmployeeNo() As Integer
        Get
            Return id
        End Get
        Set(ByVal value As Integer)
            id = value
        End Set
    End Property

    <StringLength(15, ErrorMessage:="Name Length must be between 3 and 15", MinimumLength:=3)>
    <Required>
    Public Property EmployeeName() As String
        Get
            Return name
        End Get
        Set(ByVal value As String)
            name = value
        End Set
    End Property

    <StringLength(15, ErrorMessage:="City Length must be between 3 and 15", MinimumLength:=3)>
    Public Property EmployeeCity() As String
        Get
            Return city
        End Get
        Set(ByVal value As String)
            city = value
        End Set
    End Property
    <Required>
    <Range(20000, 10000, ErrorMessage:="Salary must be between 20000 and 100000")>
    Public Property EmployeeSalary() As Integer
        Get
            Return salary
        End Get
        Set(ByVal value As Integer)
            salary = value
        End Set
    End Property
End Class
